key.core/src/main/java/de/uka/ilkd/key/macros/TryCloseMacro.java

```diff
@@ -143,6 +143,8 @@ public ProofMacroFinishedInfo applyTo(UserInterfaceControl uic, Proof proof,
             return null;
         }
 
+        proof.register(this, TryCloseMacro.class);
```

Comment:
I think this is not necessary. `#register` is for adding a resource for later use to the lifecycle of a proof.

Reply:
It is required so that the caching code knows this macro is running. When a branch is "closed" by the automatic caching search, it is left open. The macro notices this and undos all its steps, which is not desirable.